cmds/cpud/serve.go

```diff
@@ -115,6 +114,12 @@ func serve() error {
 	if err != nil {
 		return err
 	}
+
+	if *mdnsEnabled {
+		host, _ := os.Hostname()
+		mdns_server, _ := server.Advertise(host, *port, *mdnsTxt)
+		defer mdns_server.Shutdown()
```

Comment:
calling `mdns_server.Shutdown()` probably won't work if `server.Advertise` returned an error, right? So maybe you can structure the code as

```
// Comment here explaining why we will ignore the `err != nil` case.
if mdns_server, err := server.Advertise(...); err == nil {
  defer mdns_server.Shutdown()
}
```

Reply:
actually, reflecting on this a bit more, if the user specifies mdns advertisment on the command line and it goes sideways, I think we just treat it as failure.  As long as mdns advertisement isn't default, this should be fine.  I'll restructure code appropriately.